### ISSUE TITLE
OpenMPI requires both C and Fortran compilers during default spack install.

### ISSUE THREAD

**Issue by berceanu** [ComputationalRadiationPhysics/picongpu#2765]
Followind the `spack` install instructions on an `Ubuntu 18.04` box, the above message appears while building `openmpi`. It can be fixed by setting
```
   f77: /usr/bin/gfortran-7
   fc: /usr/bin/gfortran-7
```
in `compilers.yaml` or running
```
spack compiler find
```
provided of course, `gfortran` is installed.

**Comment by ax3l** [comment]
Yes, that's currently a limitation of the OpenMPI package in Spack itself. Thank you for documenting this! :sparkles:

**Comment by berceanu** [comment]
I'm not sure how important it is for building `openmpi` that `gcc` and `gfortran` should be the same version. If they should be, then the above fix will often fail, as the system `gfortran` might be a different version than the `spack` `gcc` package. So should we add `gfortran` to the spack dependency list instead?

**Comment by ax3l** [comment]
I thought that the current workflow to `spack install gcc` and then `spack compiler find` does that. Did we miss it? By default, the spack `gcc` package installs the `c`, `c++` and `fortran` languages.

**Comment by berceanu** [comment]
You're absolutely right, `gfortran` is installed along with `gcc`.

**Comment by ax3l** [comment]
Ok, so this was probably just a big mixed up while you debugged the other issues?

**Comment by berceanu** [comment]
Yes, looks like everything is working now :)

**Comment by ax3l** [comment]
Thanks for the note!
I now updated the recipe to explicitly build on OpenMPI between 1.8 and 3.99: https://github.com/ComputationalRadiationPhysics/spack-repo/commit/69327543e1892011dbbb8485a213a7c1433fa9ac

**Comment by ax3l** [comment]
@hightower8083 can you please report the GCC 5.5 build issue that you get when running `spack install openmpi@4.0.0 %gcc@5.5` upstream in https://github.com/open-mpi/ompi ? I think it's not reported yet.

**Comment by hightower8083** [comment]
sure -- i'll check it again and will tell em
